Mathlib/NumberTheory/NumberField/ClassNumber.lean

```diff
@@ -155,6 +165,32 @@ theorem isPrincipalIdealRing_of_isPrincipal_of_pow_inertiaDeg_le_of_mem_primesOv
     exact hspan ▸ inertiaDeg_pos ..
   · exact hspan ▸ hlies
 
+/-- Let `K` be a number field such that `K/ℚ` is Galois and let `M K` be the Minkowski bound of `K`.
+To show that `𝓞 K` is a PID it is enough to show that, for all (natural) primes
+`p ∈ Finset.Icc 1 ⌊(M K)⌋₊`, there is an ideal `P` above `p` such that
+either `⌊(M K)⌋₊ < p ^ (span ({p}).inertiaDeg P)` or `P` is principal. This is the standard
+technique to prove that `𝓞 K` is principal in the Galois case, see [marcus1977number], discussion
+after Theorem 37.
+
+The way this theorem should be used is to first compute `⌊(M K)⌋₊` and then to use `fin_cases`
+to deal with the finite number of primes `p` in the interval. -/
+theorem isPrincipalIdealRing_of_isPrincipal_of_lt_or_isPrincipal_of_mem_primesOver_of_mem_Icc
+    [IsGalois ℚ K] (h : ∀ p ∈ Finset.Icc 1 ⌊(M K)⌋₊, p.Prime →
+      ∃ P ∈ primesOver (span {(p : ℤ)}) (𝓞 K),
+        ⌊(M K)⌋₊ < p ^ ((span ({↑p} : Set ℤ)).inertiaDeg P) ∨
+          Submodule.IsPrincipal P) :
+      IsPrincipalIdealRing (𝓞 K) := by
+  refine isPrincipalIdealRing_of_isPrincipal_of_pow_le_of_mem_primesOver_of_mem_Icc
+    (fun p hpmem hp P ⟨hP1, hP2⟩ hple ↦ ?_)
+  obtain ⟨Q, ⟨hQ1, hQ2⟩, H⟩ := h p hpmem hp
+  have := (isPrime_of_prime (prime_span_singleton_iff.mpr (prime_iff_prime_int.mp hp))).isMaximal
```

Comment:
Note that this is an instance if you import `Mathlib.RingTheory.Ideal.Int`

Reply:
I tried, but it does not work...